kubernetes/samples/scripts/domain-lifecycle/setupPortForwarding.sh

```diff
@@ -0,0 +1,181 @@
+# !/bin/sh
+# Copyright (c) 2021, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+
+script="${BASH_SOURCE[0]}"
+scriptDir="$( cd "$( dirname "${script}" )" && pwd )"
+source ${scriptDir}/helper.sh
+if [ "${debug}" == "true" ]; then set -x; fi;
+set -eu
+
+function usage() {
+
+  cat << EOF
+
+  This script sets up the 'kubectl port-forward' for a WebLogic domain to allow access to the
+  WebLogic Administration Console. The 'kubectl port-forward' pattern is often more secure than
+  exposing the console to the public Internet. The script creates a pod in the domain namespace
+  using the image containing 'socat' binary and it forwards the request to the WebLogic
+  Admistration Server pod. The script derives the address and the adminstration port of the
+  WebLogic Admistration Server using the domain topology. The port can also be provided on the
+  command line using '-r' option.
+
+  Usage:
+
+    $(basename $0) -n namespace -d domain_uid [-a local_ip_address] [-l localPort] [-i image_name] 
+                   [-p image_pull_policy] [-w wait_timeout] [-p port_forward_server_name] [-m kubecli] 
+                   [-t pod_running_timeout]
+
+    -d <domain_uid>                : Domain unique-id. This parameter is required.
+
+    -n <namespace>                 : Domain namespace. This parameter is required.
+
+    -a <local_ip_address>          : Addresses to listen on (comma separated value for --address flag). 
+                                     Only accepts IP addresses or localhost as a value. Default is 'localhost'.
+
+    -l <local_port>                : A local port to listen on. If not specified, it listens on a random local port.
+
+    -r <remote_port>               : The port of the WebLogic Administartion Server to forward the request to.
+                                     If not specified, this script will derive the adminstration port from the
+                                     WebLogic domain topology stored in the introspector config map.
```

Comment:
The existence and nature of the "introspector config map" is not commonly known to customers.  Maybe add. "The introspector config map is automatically created by the Operator for a domain after the domain's introspector job succeeds".